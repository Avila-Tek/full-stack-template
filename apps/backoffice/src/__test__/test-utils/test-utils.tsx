import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/react';
import { QueryProvider } from '@/context/query-context';
import { QueryClientProvider } from '@tanstack/react-query';

// mock @apollo/experimental-nextjs-app-support/rsc
jest.mock('@apollo/experimental-nextjs-app-support/rsc', () => ({
  registerApolloClient: jest.fn(() => ({
    getClient: jest.fn(() => ({
      query: jest.fn(),
    })),
  })),
}));

// mock next-auth/react
jest.mock('next-auth/react', () => {
  const originalModule = jest.requireActual('next-auth/react');
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
    user: { name: 'admin' },
  };
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return { data: mockSession, status: 'authenticated' };
    }),
  };
});

// // -----------------------------------------------------------------------------
// This file re-exports everything from React Testing Library and then overrides
// its render method. In tests that require global context providers, import
// this file instead of React Testing Library.
//
// For further details, see:
// https://testing-library.com/docs/react-testing-library/setup/#custom-render
// -----------------------------------------------------------------------------

const AllProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <React.Suspense fallback={<div>Loading ...</div>}>
      <QueryProvider>{children}</QueryProvider>
    </React.Suspense>
  );
};

/**
 * Custom render method that includes global context providers
 */
type CustomRenderOptions = {
  initialRoute?: string;
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
};

function customRender(ui: React.ReactElement, options?: CustomRenderOptions) {
  const opts = options || {};
  const { initialRoute, renderOptions } = opts;

  if (initialRoute) {
    window.history.pushState({}, 'Initial Route', initialRoute);
  }

  return render(ui, { wrapper: AllProviders, ...renderOptions });
}

export * from '@testing-library/react';
export { customRender as render, userEvent };
