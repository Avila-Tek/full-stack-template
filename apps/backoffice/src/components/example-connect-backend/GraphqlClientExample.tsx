'use client';

import React from 'react';
import { useQuery } from '@apollo/client';
import { GET_USER } from '@/graphql/queries';

function GraphqlClientExample() {
  // remember to use the correct type for the data
  const { loading, data, error } = useQuery<{
    user: {
      email: string;
    };
  }>(GET_USER, {
    fetchPolicy: 'network-only',
  });

  console.log(data, loading, error);

  return <div>Hello {data?.user?.email}</div>;
}

export default GraphqlClientExample;
