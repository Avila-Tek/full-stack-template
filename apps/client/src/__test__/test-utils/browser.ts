import { setupWorker } from 'msw/browser';
import { handlers } from '../mocks/handlers';

// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...handlers);
