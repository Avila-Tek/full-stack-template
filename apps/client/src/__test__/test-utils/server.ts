import { setupServer } from 'msw/node';
import { handlers } from '../__mocks__/handlers';

// This configures a Service Worker with the given request handlers.
export const server = setupServer(...handlers);
