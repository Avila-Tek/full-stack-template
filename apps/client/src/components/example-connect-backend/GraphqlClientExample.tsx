'use client';

import React from 'react';
import { useQuery } from '@apollo/client';
import { GET_USER } from '@/graphql/queries';

function GraphqlClientExample() {
  // remember to use the correct type for the data
  const { loading, data, error } = useQuery<any>(GET_USER, {
    fetchPolicy: 'network-only',
  });

  console.log(data, loading, error);

  return <div>Hello world Hello {data?.user?.email}</div>;
}

export default GraphqlClientExample;
