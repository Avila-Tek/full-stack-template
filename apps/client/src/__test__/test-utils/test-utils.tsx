import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { worker } from './browser';
import userEvent from '@testing-library/user-event';

// ----- Set up Mock Service Worker -----

// Establish API mocking before all tests.
beforeAll(() => worker.start());

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => worker.resetHandlers());

// Clean up after the tests are finished.
afterAll(() => worker.stop());

// -----------------------------------------------------------------------------
// This file re-exports everything from React Testing Library and then overrides
// its render method. In tests that require global context providers, import
// this file instead of React Testing Library.
//
// For further details, see:
// https://testing-library.com/docs/react-testing-library/setup/#custom-render
// -----------------------------------------------------------------------------

const AllProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <React.Suspense fallback={<div>Loading ...</div>}>
      {children}
    </React.Suspense>
  );
};

/**
 * Custom render method that includes global context providers
 */
type CustomRenderOptions = {
  initialRoute?: string;
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
};

function customRender(ui: React.ReactElement, options?: CustomRenderOptions) {
  const opts = options || {};
  const { initialRoute, renderOptions } = opts;

  if (initialRoute) {
    window.history.pushState({}, 'Initial Route', initialRoute);
  }

  return render(ui, { wrapper: AllProviders, ...renderOptions });
}

export * from '@testing-library/react';
export { customRender as render, userEvent };
