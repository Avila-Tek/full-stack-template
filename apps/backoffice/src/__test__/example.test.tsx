import React from 'react';
import Page from '../app/page';
import { render, waitFor, screen } from './test-utils/test-utils';
import GraphqlClientExample from '@/components/example-connect-backend/GraphqlClientExample';

// Mock the Google Font loader
jest.mock('next/font/google', () => ({
  Inter: jest.fn(() => 'MockedInter'),
}));

it('renders homepage unchanged', () => {
  const { container } = render(<Page />);

  expect(container).toMatchSnapshot();
});

// describe('Graphql', () => {
//   it('renders user data correctly', async () => {
//     render(<GraphqlClientExample />);

//     // Espera a que los datos de los usuarios se rendericen
//     await waitFor(() => {
//       expect(screen.getByText('juanperez@example.com')).toBeInTheDocument();
//     });
//   });
// });
