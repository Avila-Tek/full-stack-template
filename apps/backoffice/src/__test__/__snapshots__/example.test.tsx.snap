// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    Home
    <div>
      UserInfo: null
    </div>
    <a
      href="/sign-in"
    >
      Sign In
    </a>
  </div>
</div>
`;
